spot_rl_experiments/utils/hab3_policy_conversion.py

```diff
@@ -0,0 +1,341 @@
+
+        # Using torch script to save action distribution
+        traced_cell_ad = torch.jit.trace(
+            self.policy.action_distribution.mu_maybe_std,
+            (torch.tensor(torch.zeros((1, 512))).to(self.device)),
+            strict=False,
+        )
+        torch.jit.save(traced_cell_ad, save_path["action_distribution"])
+
+        # Using torch script to save std
+        torch.save(self.policy.action_distribution.std, save_path["std"])
+
+        print("Save (1) net (2) action distribution (3) std")
+
+        self.ts_net = torch.jit.load(save_path["net"], map_location=str(self.device))
+        self.ts_action_dis = torch.jit.load(
+            save_path["action_distribution"], map_location=str(self.device)
+        )
+        self.std = torch.load(save_path["std"], map_location=str(self.device))
+        print("Load the torchscrip policy successfully")
+
+    def act(self, observations):
+        """Use the noraml way to get action"""
+        assert self.reset_ran, "You need to call .reset() on the policy first."
+        batch = batch_obs([observations], device=self.device)
+        with torch.no_grad():
+            PolicyActionData = self.policy.act(
+                batch,
+                self.test_recurrent_hidden_states,
+                self.prev_actions,
+                self.not_done_masks,
+                deterministic=True,
+            )
+            actions = PolicyActionData.actions
+            self.test_recurrent_hidden_states = PolicyActionData.rnn_hidden_states
+            # print("original hidden state:", PolicyActionData.rnn_hidden_states)
+
+        self.prev_actions.copy_(actions)
+        self.not_done_masks = torch.ones(1, 1, dtype=torch.bool, device=self.device)
+
+        # GPU/CPU torch tensor -> numpy
+        actions = actions.squeeze().cpu().numpy()
+
+        return actions
+
+    def get_action(self, mu_maybe_std, std):
+        """Small wrapper to get the final action"""
+        mu_maybe_std = mu_maybe_std.float()
+        mu = mu_maybe_std
+
+        mu = torch.tanh(mu)
+        std = torch.clamp(std, -5, 2)
+        std = torch.exp(std)
+
+        return CustomNormal(mu, std, validate_args=False)
+
+    def act_ts(self, observations):
+        """Using torchscript to run the model"""
+        assert self.ts_net is not None, "Please load the torchscript policy first!"
+
+        # Using torch script to save the model
+        batch = batch_obs([observations], device=self.device)
+        with torch.no_grad():
+            output_model = self.ts_net(
+                batch,
+                self.test_recurrent_hidden_states,
+                self.prev_actions,
+                self.not_done_masks,
+            )
+
+        features = output_model[0]
+        rnn_hidden_states = output_model[1]
+
+        with torch.no_grad():
+            action = self.get_action(self.ts_action_dis(features), self.std)
+            action = action.mean
+        return action, rnn_hidden_states
+
+
```

Comment:
@jimmytyyang 
Let's also add a documentation outlining the structure for all policy converter classes that will be defined in future.

Reply:
@jimmytyyang  addressed , will reflect in next commit